cli/src/commands/operation.rs

```diff
@@ -75,6 +81,19 @@ pub struct OperationUndoArgs {
     what: Vec<UndoWhatToRestore>,
 }
 
+/// Prune unneeded operations (EXPERIMENTAL)
+#[derive(clap::Args, Clone, Debug)]
+pub struct OperationPruneArgs {
+    /// The latest operation to prune
+    ///
+    /// The specified operation and its all ancestors will be removed. The
+    /// descendant operations will have new ids.
+    #[arg(long, required = true)]
+    ancestors: Option<String>,
+    // TODO: There may be an option to move op heads backwards to discard recent
+    // commit objects?
```

Comment:
FWIW, I have been thinking of adding a `jj op abandon` for that. But I haven't even started working on it.